Polyfills/Window/Source/Window.cpp

```diff
@@ -32,46 +40,85 @@ namespace
             , m_thread{&TimeoutDispatcher::WaitThenCallProc, this}
         { }
 
-        void Dispatch(std::shared_ptr<Napi::FunctionReference> func, std::chrono::system_clock::time_point time) 
-        {   
-            if (time <= std::chrono::system_clock::now())
+        TimeoutId Dispatch(std::shared_ptr<Napi::FunctionReference> func, Milliseconds time)
+        {
+            if (time <= 0)
             {
                 CallFunction(std::move(func));
-                return;
+                return 0;
             }
 
+            const auto now = Now();
             std::unique_lock<std::mutex> lk(m_mutex);
             const auto soonestTimePoint =
-                m_queue.empty()
-                    ? time + std::chrono::milliseconds{1}
-                    : m_queue.top().TimePoint;
+                m_timeMap.empty()
+                ? now + std::chrono::milliseconds{ 1 }
+            : (*m_timeMap.cbegin()).second.front().timePoint;
```

Comment:
Generally, all over the file, you can use `iterator->` syntax instead of `(*iterator).`